libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -59,4 +61,12 @@ export class BookSearchComponent implements OnInit {
       this.store.dispatch(clearSearch());
     }
   }
+
+  throttleSearchQuery(event) {
+    from(event)
+    .pipe(throttleTime(500))
```

Comment:
Have you looked at the console while this runs? If you type `Javascript` into the box it makes a call for every letter and throws proxy errors. This almost certainly isn't what you want.

Look into `FormControl.valueChanges` and check out `debounceTime` and maybe even `distinctUntilChanged`.

Reply:
updated